api_tests/preprints/views/test_preprint_detail.py

```diff
@@ -94,6 +103,39 @@ def test_embed_contributors(self, app, user, preprint):
         for contrib in embeds['contributors']['data']:
             assert contrib['id'] in ids
 
+    def test_preprint_doi_link_absent_in_unpublished_preprints(self, app, user, unpublished_preprint, unpublished_url):
+        res = app.get(unpublished_url, auth=user.auth)
+        assert res.json['data']['id'] == unpublished_preprint._id
+        assert res.json['data']['attributes']['is_published'] == False
+        assert 'self' in res.json['data']['links'].keys()
+        assert 'html' in res.json['data']['links'].keys()
```

Comment:
No need to duplicate these asserts across tests. Better to have the tests be focused on a few things so that when they fail, it's more obvious where to look for the bug.